js/Regex Password Validation.js

```diff
@@ -0,0 +1,7 @@
+// https://www.codewars.com/kata/52e1476c8147a7547a000811/train/javascript
+
+//Not sure what ist wrong but in 3 cases test not pass.
+
+function validate(password) {
+    return /(?=.*\d)(?=.*[a-z])(?=.*[A-Z]){6,6}/.test(password);
```

Comment:
I'm not an expert on regexps :smile: The only thing I see that could be improved is that the value is supposed to be at least 6 characters, so
```suggestion
    return /(?=.*\d)(?=.*[a-z])(?=.*[A-Z]){6}/.test(password);
```

could be a bit more correct in this case, rather than `{6,6}` which matches exactly 6 characters.

Aside from that, I would not use a single regexp to test match those criteria :smile: I would use 3 individual ones (one for lowercase, one for uppercase, one for a digit) and check the length of the string.

I know that this kata suggested using a regexp, but you can try to use another approach and then see other people's results :smile: